deploy-clojars/action.yml

```diff
@@ -0,0 +1,89 @@
+name: Clojars Deployment
+
+description: Build JAR and Deploy to Clojars
+
+inputs:
+  group-id:
+    description: The Clojars Group ID. Defaults to 'com.yetanalytics'.
+    required: false
+    default: 'com.yetanalytics'
+
+  artifact-id:
+    description: The Clojars Artifact ID (i.e. the name of the lib itself)
+    required: true
+
+  version:
+    description: >
+      The version string. Defaults to the Git branch or tag associated with
+      the event that triggered this action. NOTE: If the tag is prefixed
+      (e.g. 'v1.0.0'), you should remove the prefix, then pass in the
+      truncated tag as the arg (e.g. pass in `${GITHUB_REF#refs\/tags\/v}`).
+    required: false
+    default: ${{ github.ref_name }}
+
+  src-dirs:
+    description: >
+      A quoted array of the source directories. Defaults to '["src/main"]'.
+    required: false
+    default: '["src/main"]'
+
+  resource-dirs:
+    description: >
+      A quoted array of the resource directories. Defaults to '["resources"]'.
+      Pass '[]' if no resource directories exist.
+    required: false
+    default: '["resources"]'
+
+  # GitHub Secrets
+
+  clojars-username:
+    description: The Clojars username (should be stored as a GitHub secret).
+    required: true
+
+  clojars-password:
+    description: The Clojars password (should be an access token stored as a GitHub secret).
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Add aliases
+      shell: bash
+      run: >
+        clojure -Sdeps
+        '{:aliases
+          {:build
+           {:replace-deps {io.github.seancorfield/build-clj {:git/tag "v0.6.3" :git/sha "9b8e09b"}}
+            :ns-default build}}}'
```

Comment:
Perhaps I misunderstand the nature of the -Sdeps arg... I thought it just merges into the deps.edn for _that specific invocation_, does it actually write to the deps.edn file?

Reply:
What makes you think that?

Reply:
The docs? For instance, running what you have here `-Sdeps` and a string with edn for me, opens a repl. It doesn't make sense as a build step on its own, I believe the `-Sdeps ...` has to be present on any call where you use the alias `:build`

Reply:
You might be right - I've been testing this with the `:build` alias still present in colossal-squuid's deps.edn file, so it's likely that if I remove it this would break.

Reply:
right, if this works it is because `:build` is present in the calling project and happens to have the same stuff